src/devices/Ssd1351/Command/Commands.cs

```diff
@@ -0,0 +1,56 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Text;
+
+namespace Iot.Device.Ssd1351.Command
+{
+    public enum SSD1351Command : byte
+    {
+        SSD1351_CMD_SETCOLUMN = 0x15, // used
+        SSD1351_CMD_SETROW = 0x75, // used
+        SSD1351_CMD_WRITERAM = 0x5C, //used
+        SSD1351_CMD_READRAM = 0x5D,
+        SSD1351_CMD_SETREMAP = 0xA0, //used
+        SSD1351_CMD_STARTLINE = 0xA1, //used
+        SSD1351_CMD_DISPLAYOFFSET = 0xA2, //used
+        SSD1351_CMD_DISPLAYALLOFF = 0xA4,
+        SSD1351_CMD_DISPLAYALLON = 0xA5,
+        SSD1351_CMD_NORMALDISPLAY = 0xA6, //used
+        SSD1351_CMD_INVERTDISPLAY = 0xA7, //used
+        SSD1351_CMD_FUNCTIONSELECT = 0xAB, //used
+        SSD1351_CMD_DISPLAYOFF = 0xAE, //used
+        SSD1351_CMD_DISPLAYON = 0xAF, //used 
+        SSD1351_CMD_PRECHARGE = 0xB1, //used
+        SSD1351_CMD_DISPLAYENHANCE = 0xB2,
+        SSD1351_CMD_CLOCKDIV = 0xB3, //used
+        SSD1351_CMD_SETVSL = 0xB4, //used 
+        SSD1351_CMD_SETGPIO = 0xB5, //used
+        SSD1351_CMD_PRECHARGE2 = 0xB6, //used
+        SSD1351_CMD_SETGRAY = 0xB8,
+        SSD1351_CMD_USELUT = 0xB9,
+        SSD1351_CMD_PRECHARGELEVEL = 0xBB,
+        SSD1351_CMD_VCOMH = 0xBE,  //used 
+        SSD1351_CMD_CONTRASTABC = 0xC1, //used
+        SSD1351_CMD_CONTRASTMASTER = 0xC7, //used
+        SSD1351_CMD_MUXRATIO = 0xCA, //used
+        SSD1351_CMD_COMMANDLOCK = 0xFD, // used
+        SSD1351_CMD_HORIZSCROLL = 0x96,
+        SSD1351_CMD_STOPSCROLL = 0x9E,
+        SSD1351_CMD_STARTSCROLL = 0x9F,
+    }
+
+    public interface ICommand
```

Comment:
Does having this interface buys anything more rather than having methods directly on Ssd1351. I.e. something along the lines of

```csharp
private void SendCommand(Command cmd, ReadOnlySpan<byte> additionalData)
{
  if (additionalData.Length > MaxSpanLengthInExample64Bytes)
    throw new Exception("TODO: regular allocation or change to Debug.Assert if not needed");
  Span<byte> buff = stackalloc byte[additionalData.Length + 1];
  buff[0] = (byte)cmd;
  additionalData.CopyTo(buff.Slice(1));
  _spiOrI2c.Write(buff);
}

private void SendCommand(Command cmd, byte additionalData)
{
  Span<byte> data = stackalloc byte[1];
  data[0] = additionalData;
  SendCommand(cmd, data);
}
public void Lock(bool lockState)
{
  SendCommand(Command.Lock, lockState ? 0x12 : 0x16); // or whatever the values were
}
```

Reply:
Well I sort of agree with you regarding this but I was trying to follow the way that the existing Ssd1306 was engineered which uses an ICommand interface. I would be happy to refactor if that is what is needed. Please confirm which way you would like me to go,

Reply:
In general it's rather matter of preference, internally I've rather seen people pushing back on the interfaces. We have allowed some of that in the past reviews since it's much better to have a support for a new device than not having it and it seemed like quite a bit of work to refactor that.

Before doing any work I'd recommend creating a separate issue to decide which direction people think we should go (please cc at minimum everyone which added/modified previous bindings, me and @joperezr).

For this PR you can leave as is or if you think the non-interface version feels better then go ahead and refactor only this code so that we can compare usage before & after and decide which one is better (note that implementation detail is less important than user experience).

Note that we are still trying to establish generic patterns in this repo and therefore we don't have specific guidelines for this case (so far most of the devices do not use ICommand interface or any similar concept and rather directly expose properties or methos but Ssd1306 do).

Most common and generally (IMO) easier to use pattern would be to have properties (i.e. instead of method Lock I shared above)

```csharp
bool Locked
{
  get
  {
    // TODO: either read cached value or if possible ask hardware what is the true state
  }
  set
  {
    SendCommand(Command.Lock, value ? 0x12 : 0x16); // or whatever the values were
  }
}
```

this way you can write code like following:

```csharp
ssd1351.Locked = true;
ssd1351.Locked = false;
```

vs

```
ssd1351.SendCommand(new LockCommand());
ssd1351.SendCommand(new UnlockCommand());
```

except for syntax being easier and more discoverable (since intellisense will help you with list of available commands when you type `ssd1351.`) you also don't need to allocate the command (although this is arguable since you can cache all popular commands)

Reply:
> ssd1351.Locked = true;
> ssd1351.Locked = false;

While this does make it cleaner sometimes, I think this type should stay in method format... especially since it is related to processing data via protocol for the binding device to update hardware.  It will confuse binding creators on when/not to use. For example, one might code up...

```csharp
// used as property
ssd1351.Lock = true;
// and for another command as method
ssd1351.SetColumnAddress(0x00, 0x7F);
```

Below shows in method format...
```csharp
ssd1351.Lock();
// or could use a flag to perform Lock and Unlock commands.
ssd1351.Lock(true);
ssd1351.Lock(false);
```

I like how we used a property to control a pin where it was a master GPIO controller that interacted with the binding pins directly and not through the device.  Use [Mcp25xxx pin](https://github.com/dotnet/iot/blob/26e1b07557ce650d3110769262a27b50f38650be/src/devices/Mcp25xxx/Mcp25xxx.cs#L124) as example.

So if the Ssd1351 device had control pins, like Output Enable or Lock, I could see being used as a property.

```csharp
ssd1351.Lock = true;  // or ssd1351.Lock = PinValue.High;
ssd1351.OutputEnable = false;
```

This might be another detail we can add in guidance related to #463.